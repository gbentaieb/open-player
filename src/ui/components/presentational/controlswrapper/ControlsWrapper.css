.ControlsWrapper {
  position: absolute;
  display: flex;
  text-align: center;
  justify-content: center;
  bottom: 0;
  padding: 30px 5px 5px 5px;
  height: 55px;
  width: calc(100% - 10px);
  z-index: 10;
  background-image: linear-gradient(to top, rgba(0, 0, 0, 0.5), transparent);
  opacity: 0;
  transition: opacity 0.2s ease-in-out;
}

.ControlsWrapperActive {
  opacity: 1;
}
