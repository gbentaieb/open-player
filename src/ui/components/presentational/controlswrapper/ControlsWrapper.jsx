import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import style from './ControlsWrapper.css';

class ControlsWrapper extends Component {
  static propTypes = {
    isActive: PropTypes.bool.isRequired,
    children: PropTypes.node,
  }

  static defaultProps = {
    children: <div />,
  }

  render() {
    const className = classNames({
      [style.ControlsWrapper]: true,
      [style.ControlsWrapperActive]: this.props.isActive,
    });

    return (
      <div className={className}>
        {this.props.children}
      </div>
    );
  }
}

export default ControlsWrapper;
