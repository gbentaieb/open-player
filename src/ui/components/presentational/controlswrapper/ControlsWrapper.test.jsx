import React from 'react';
import ControlsWrapper from './ControlsWrapper';

const defaultProps = {
  isActive: true,
};

describe('ControlsWrapper snapshot', () => {
  test('ControlsWrapper > Snapshot', () => {
    const wrapper = shallow(<ControlsWrapper {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
