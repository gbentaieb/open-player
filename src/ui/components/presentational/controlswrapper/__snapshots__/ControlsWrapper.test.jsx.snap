// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsWrapper snapshot ControlsWrapper > Snapshot 1`] = `
<div
  className="undefined"
>
  <div />
</div>
`;
